clients/vscode/src/chat/ChatViewProvider.ts

```diff
@@ -128,6 +130,50 @@ export class ChatViewProvider implements WebviewViewProvider {
 
         this.sendMessage(chatMessage);
       },
+      onApplyInEditor: (content: string) => {
+        const editor = window.activeTextEditor;
+        if (editor) {
+          const document = editor.document;
+          const selection = editor.selection;
+          const text = editor.document.getText(selection);
+          const { start } = selection;
+
+          // Find the minimum indent
+          const lineText = document.lineAt(start.line).text;
+          const selectedLines = text.split("\n");
+          const allLines = [lineText].concat(selectedLines.slice(1));
+          let subsequentLinesMinIndent = "";
+          let subsequentLinesMinIndentLength = Infinity;
+          allLines.forEach((line) => {
+            const match = line.match(/^(\s*)/);
+            const indent = match ? match[0] : "";
+            if (indent.length < subsequentLinesMinIndentLength) {
+              subsequentLinesMinIndent = indent;
+              subsequentLinesMinIndentLength = indent.length;
+            }
+          });
+
+          // Add the indent at the beginning of the content
+          const indentedContent = content
+            .split("\n")
+            .map((line) => subsequentLinesMinIndent + line)
+            .join("\n");
+
+          // Apply into the editor
+          editor.edit((editBuilder) => {
+            const { start, end } = selection;
+            const startBeginPosition = new Position(start.line, 0);
+            const hasSelectionInEditor = text.length > 0;
+
+            if (hasSelectionInEditor) {
+              const adjustedRange = new Range(startBeginPosition, end);
+              editBuilder.replace(adjustedRange, indentedContent);
+            } else {
+              editBuilder.insert(startBeginPosition, indentedContent);
```

Comment:
If there is no selection, using `replace` seems also works.

Reply:
correct